benchmarks/bench_feature_expansions.py

```diff
@@ -0,0 +1,52 @@
+import matplotlib.pyplot as plt
+import numpy as np
+import scipy.sparse as sparse
+from sklearn.preprocessing import PolynomialFeatures
+from sys import argv
+from time import time
+
+if len(argv) < 2:
+    print("Please specify where to save the resulting plot "
+          "via the a command line param.")
+    exit()
+
+degree = 2
+trials = 5
+num_rows = 100
+densities = [0.1, 0.4, 0.7, 1.0]
+dimensionalities = [100, 200, 400, 800, 1600]
+dense_times = {d: np.zeros(len(dimensionalities)) for d in densities}
+sparse_times = {d: np.zeros(len(dimensionalities)) for d in densities}
+transform = PolynomialFeatures(degree=degree, include_bias=False,
+                               interaction_only=False)
+
+for trial in range(trials):
+    for density in densities:
+        for dim_index, dim in enumerate(dimensionalities):
+            print(trial, density, dim)
+            X_csr = sparse.random(num_rows, dim, density).tocsr()
+            X_dense = X_csr.toarray()
```

Comment:
Also could you please include `X_csc = X_csr.tocsc()` in this benchmark?

Reply:
Sure, but csc is crazy slow, so I'm adding a subplot where I compare csc and csr. Otherwise dense and csr would be almost indistinguishable because csc's times would squash everything. Sound OK?

Reply:
Maybe use a log scale (and do not compute CSC for the largest problem to keep the overall runtime of the benchmark under control).

Reply:
How's that looking?

https://drive.google.com/file/d/1IUZSsP_mdz5UkVx-Mfi4kte2Jd0yT-Bo/view

Reply:
Thanks! Note that it's possible to directly report plot in the github discussion:

![image](https://user-images.githubusercontent.com/89061/46336439-14d1fe80-c62b-11e8-8d3f-06d06b4c1762.png)

Maybe we could have done one sub plot per density level but this is nitpicking. I think the point is that the new CSR-based implementation is significantly faster than both the dense and the CSC variants.

Reply:
Interesting difference between our machines.

I did implement a Cython version for dense when I wrote the paper, just so that the comparison would be fair. I found that dense started off faster, but as D grew and d shrank, sparse became faster, as one would expect.

Reply:
Here's the loglog plot:
![image](https://user-images.githubusercontent.com/1478067/46395169-6072bf80-c6a0-11e8-80cd-63b76ef40e4c.png)

Reply:
Regarding separating the plots into subplots per density, since only the csr method varies with density, each subplot would be pretty similar. Having them all on the same plot with an alpha < 1 kind of shows that only the csr method varies with density, which is partially what I wanted to show. Let me know if you want the change though.

Reply:
Sorry I had not seen these comments. I think the fact that it was faster than dense was because we ran benchmarks on a very low number of samples which was not representative of realistic situation. The new setting is more interesting. I think.